boards: arm: Add support for STM32H7B3I_DK board and STM32H7B3 SoCs (#41981)
zephyrproject-rtos/zephyr: boards/arm/stm32h7b3i_dk/doc/index.rst

| PINMUX    | on-chip    | pinmux                              |
+-----------+------------+-------------------------------------+
| GPIO      | on-chip    | gpio                                |
+-----------+------------+-------------------------------------+
| I2C       | on-chip    | i2c                                 |
+-----------+------------+-------------------------------------+
| FLASH     | on-chip    | flash memory                        |
+-----------+------------+-------------------------------------+
| FMC       | on-chip    | memc (SDRAM)                        |
+-----------+------------+-------------------------------------+

Other hardware features are not yet supported on Zephyr porting.

The default configuration per core can be found in the defconfig file:
``boards/arm/stm32h7b3i_dk/stm32h7b3i_dk_defconfig``

Pin Mapping
===========

For mode details please refer to `STM32H7B3I-DK website`_.

Default Zephyr Peripheral Mapping:
----------------------------------

- UART_1 TX/RX : PA9/PA10 (ST-Link Virtual Port Com)
- UART_4 TX/RX : PB9/PB8 (Arduino Serial)
- LD1 : PG11
- LD2 : PG2
- USER_PB : PC13

> 

Reviewer: Would be nice to add FMC here. If too much tedious, let's point to board dts file as reference

+-----------+------------+-------------------------------------+
| Interface | Controller | Driver/Component                    |
+===========+============+=====================================+
| NVIC      | on-chip    | nested vector interrupt controller  |
+-----------+------------+-------------------------------------+
| UART      | on-chip    | serial port-polling;                |
|           |            | serial port-interrupt               |
+-----------+------------+-------------------------------------+
| PINMUX    | on-chip    | pinmux                              |
+-----------+------------+-------------------------------------+
| GPIO      | on-chip    | gpio                                |
+-----------+------------+-------------------------------------+
| I2C       | on-chip    | i2c                                 |
+-----------+------------+-------------------------------------+
| FLASH     | on-chip    | flash memory                        |
+-----------+------------+-------------------------------------+
| FMC       | on-chip    | memc (SDRAM)                        |
+-----------+------------+-------------------------------------+

> Other hardware features are not yet supported on Zephyr porting.

Reviewer 2: ```suggestion
Other hardware features have not been enabled yet for this board.
```
To be consistent with #41679

Author: Changed

Reviewer 2: Thanks. Please squash the fix into the original commit that introduced this file.

Author: Squashed and rebased

Thank you for the quick feedback
